dataverse/webapi/C#-NETx/BulkOperations/UpsertMultiple/Program.cs

```diff
@@ -0,0 +1,349 @@
+                    Console.WriteLine($"\nPreparing {num} records to Upsert(create) using Primary Key..");
+
+                    for (int i = 0; i < num; i++)
+                    {
+                        entityList.Add(new JObject() {
+                            {"sample_name", $"sample record {i+1:0000000}"},
+                            {tablePrimaryKeyName, Guid.NewGuid() },
+                            // Each record MUST have the @odata.type specified.
+                            {"@odata.type",$"Microsoft.Dynamics.CRM.{tableLogicalName}" }
+                        });
+                    }
+
+                    // 2. Create numberOfRecords/4 records to be created using Alternate Key
+                    if (Settings.CreateAlternateKey)
+                    {
+                        Console.WriteLine($"\nPreparing {num} records to Upsert(create) using Alternate Key..");
+                        for (int i = (int)num; i < numRecUpsertCreate; i++)
+                        {
+                            if (Settings.UseElastic)
+                            {
+                                entityList.Add(new JObject() {
+                            { "sample_name", $"sample record {i+1:0000000}"},
+                            { "@odata.id", $"{tableLogicalName}s({tablePrimaryKeyName}={Guid.NewGuid()},{elasticTablePartitionId}='sample record upsert record key {i + 1:0000000}')" },
+                            // Each record MUST have the @odata.type specified.
+                            {"@odata.type",$"Microsoft.Dynamics.CRM.{tableLogicalName}" }
+                        });
+                            }
+                            else
+                            {
+                                entityList.Add(new JObject() {
+                            {"sample_name", $"sample record {i+1:0000000}"},
+                            { "@odata.id", $"{tableLogicalName}s({tableAlternateKey}='sample record upsert record key {i+1:0000000}')" },
+                            // Each record MUST have the @odata.type specified.
+                            {"@odata.type",$"Microsoft.Dynamics.CRM.{tableLogicalName}" }
+                        });
+                            }
+                        }
+                    }
+
+                    // Use UpsertMultipleRequest
+                    UpsertMultipleRequest upsertMultipleRequest2 = new(
+                        entitySetName: tableSetName,
+                        targets: entityList);
+
+                    // Add a tag optional parameter to set a shared variable to be available to a plug-in.
+                    // See https://learn.microsoft.com/power-apps/developer/data-platform/optional-parameters?tabs=webapi#add-a-shared-variable-to-the-plugin-execution-context
+
+                    // TODO?: Add a 'tag' parameter to the UpsertMultipleRequest constructor provide the option to set this
```

Comment:
As mentioned above, this is just an idea. Not essential.

Reply:
Done